knowhere.go

```diff
@@ -0,0 +1,83 @@
+package knowhere
+
+/*
+#cgo pkg-config: knowhere
+#include <stdlib.h>
+int CheckConfig(int index_type, char const* str, int n, int param_type);
+*/
+import "C"
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"unsafe"
+)
+
+func CheckConfig(name string, params map[string]string, ptype string) error {
```

Comment:
Suggest to put the cgo function in segcore aligning with all other APIs.
We can put the `IndexType - number` mapping in Knowhere side.